syntax = "proto3";

/**********************************************************
*
*协议文件定义规范, {Module}为模块枚举定义
*
*package ProtoMsg/Msg{Module};
*
*message MsgModule{Module}
*{
	enum Cmd
	{
		Msg_{Module}_XXX_Cs 		= 1;
		Msg_{Module}_XXX_Sc			= 2;
		Msg_{Module}_XXX_Notify     = 3;
	}
}
**********************************************************/

package ProtoMsg;
//网络消息模块类型
enum MsgModule 
{
	Begin				= 0;		
	Mapping				= 1;		//网络模块开始,用作注册CmdMapping
	Login				= 2;		//登录
	
	Hero				= 3;		//英雄角色
	Friend				= 4;		//好友
	Mall				= 5;		//商城
	Mail				= 6;		//邮件
	Faction				= 7;		//工会
	Active				= 8;		//活动
	Ranker				= 9;		//排行榜

	Battle				= 80;		//战斗
	ServerCommon        = 90;		
	ServerInner			= 91;		//服务器之间消息
	ServerDB			= 92;		//服务器和数据服消息
	ServerBattle		= 93;		//服务器和战斗服消息

	End					= 100;		


}